### PR TITLE
feat: `forRoot`

### PR DESCRIPTION
**Pull request by player31-kks**
### Add importSync
I created a synchronous file reading function to create `forRoot` function.

* create forRoot function 
* create ImporSync class

### Troubleshooting when introducing other dependencies.
```ts
Module({
      ...originalMetadata,
      imports: [...originalMetadata.imports],
      controllers: [...originalMetadata.controllers, ...(module.controllers ?? [])],
      providers: [...originalMetadata.providers, ...(module.providers ?? [])],
    })(target);
  };
```
I imported the origin module as it is, and put the remaining `autoInjectable` and `autoController` in the origin module

## References

no References

## Checklist

- [x] Add labels and reviewers ?
- [x] Are you fully explaining what you changed ?
- [x] Did you check that there is no conflict with the base branch ?

### CODE REVIEW

**Comment by Nhahan** [comment]
This PR will mark a magnificent stride for our project. 😆

**Comment by Nhahan** [comment]
This PR will mark a magnificent step for our project. 😆

File: lib/auto.module.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { DynamicModule, Module } from '@nestjs/common';
 import { Importer } from './core';
+import { ImporterSync } from './core/importSync';
 
 @Module({})
```
[the remark this answers is not in the document]
Comment (Nhahan) [reply lib/auto.module.ts:5]: Sorry. It's my IDE bug. (`WebStorm` sucks)

File: lib/core/index.ts [hunk 2]
```diff
@@ -1 +1,2 @@
 export * from './importer';
+export * from './importSync';
```
[the remark this answers is not in the document]
Comment (player31-kks) [reply lib/core/index.ts:2]: thanks i'll change

File: lib/auto.module.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import { DynamicModule, Module } from '@nestjs/common';
 import { Importer } from './core';
+import { ImporterSync } from './core/importSync';
```
[the remark this answers is not in the document]
Comment (player31-kks) [reply lib/auto.module.ts:3]: Thanks for the suggestion.

File: lib/core/importSync.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+type ClassType = new (...args: any[]) => any;
+
+interface AutoClasses {
+  providers: ClassType[];
+  controllers: ClassType[];
+}
+
+export class ImporterSync {
+  private static instance: ImporterSync | null = null;
```
[the remark this answers is not in the document]
Comment (player31-kks) [reply lib/core/importSync.ts:15]: I did bring it over from the Importer, but I will make the necessary modifications!

File: lib/core/importSync.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
```
[the remark this answers is not in the document]
Comment (Nhahan) [reply lib/core/importSync.ts]: nevermind

**Review by Nhahan — APPROVED** [review]
the Best PR ever. really.

**Pull request merged**